### PR TITLE
feat(deployment): tool to create CircleCI jobs easily and integration…

### PR DESCRIPTION
**Pull request by jlandure**
… in one appengine project: zen-formations. Have to try with real use cases 🦄

### CODE REVIEW

File: ic.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+const assert = require('assert');
+const fs = require('fs');
+const rp = require('request-promise-native');
+const username = process.env.CIRCLE_TOKEN;
+const password = '';
+const auth = 'Basic ' + new Buffer(username + ':' + password).toString('base64');
+
+var projectName = JSON.parse(fs.readFileSync('./package.json')).name;
+//Override project name in the command line
+if(process.argv.length === 3) projectName = process.argv[2]
```
Comment (hgwood) [inline ic.js]: What do you think of `const projectName = process.argv[2] || require('./package.json').name`?

File: ic.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    .then(function (data) {
+        return rp(lastGetFirstGreenBuild)
+    })
+    .then(function (data) {
+        return console.log(`💚 All is done! Wait for a green deployment`)
+    })
+    .catch(function (err) {
+      console.log('💩 AieAieAie!\n', err)
+        // API call failed...
+    });
```
Comment (hgwood) [inline ic.js]: - Arrow functions might be useful here.
- Might be clearer to drop the callback parameter `data` when the previous callback does not return anything.
- Might be less surprising to drop `return`s in front of `console.log`s
- Nice use of emojis ! 🥇

File: ic.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      'Authorization': auth
+    },
+    body: {
+        name: 'GAE_SERVICE_ACCOUNT',
+        value: process.env.GAE_SERVICE_ACCOUNT,
+    },
+    json: true,
+};
+
+const thirdSetEnvVariable2 = {
```
Comment (hgwood) [inline ic.js]: Might be clearer with "1" and "2" replaced with meaningful name.

File: ic.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+const secondCreateProject = {
+    method: 'POST',
+    uri: `https://circleci.com/api/v1/project/Zenika/${projectName}/follow`,
+    headers: {
+      'Authorization': auth
+    },
+    json: true, // Automatically parses the JSON string in the response
+};
+
+const thirdSetEnvVariable1 = {
```
Comment (hgwood) [inline ic.js]: Not sure if putting the order in the same is useful or not. 🤔

File: ic.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env node
+
+const assert = require('assert');
+const fs = require('fs');
+const rp = require('request-promise-native');
```
Comment (hgwood) [inline ic.js]: Might be clearer for later code to call this variable `request`.

File: ic.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    uri: `https://circleci.com/api/v1/project/Zenika/${projectName}/1/retry`,
+    headers: {
+      'Authorization': auth
+    },
+    json: true,
+}
+
+rp(firstCheckToken)
+    .then(function (data) {
+        assert(data.login, "jlandure")
```
Comment (hgwood) [inline ic.js]: Looks suspicious 🔍

File: ic.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    resolveWithFullResponse: false,
+};
+
+const secondCreateProject = {
+    method: 'POST',
+    uri: `https://circleci.com/api/v1/project/Zenika/${projectName}/follow`,
+    headers: {
+      'Authorization': auth
+    },
+    json: true, // Automatically parses the JSON string in the response
```
Comment (hgwood) [inline ic.js]: The comment doesn't sound too useful, as this info is easily discovered in the `request` doc.

File: ic.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+};
+
+const lastGetFirstGreenBuild = {
+    method: 'POST',
+    uri: `https://circleci.com/api/v1/project/Zenika/${projectName}/1/retry`,
+    headers: {
+      'Authorization': auth
+    },
+    json: true,
+}
```
Comment (hgwood) [inline ic.js]: Can't see where the CircleCI project is linked to a GitHub repo?

File: ic.js (hunk 1 above)
Comment (jlandure) [inline ic.js]: j'ai hésité ! mais ça me va bien.

File: ic.js (hunk 8 above)
Comment (jlandure) [inline ic.js]: In the first call, with the `follow`url.

File: README.md [hunk 9]
```diff
@@ -48,6 +50,43 @@ Les exercices de TP sont écrits également en markdown, puis convertis en PDF g
...
+```
+
+### ic.js
+
+Permet d'initialiser un projet CircleCI à partir d'un repository GitHub.
+Nécessite 3 variables d'environnement:
+```
+CIRCLE_TOKEN=token circle
+GAE_SERVICE_ACCOUNT=email de service
+GAE_KEY_FILE_CONTENT=clé au format json
```
Comment (hgwood) [inline README.md]: Peut-être qu'il faudrait expliquer comment on obtient ces valeurs ?

**Comment by hgwood** [comment]
All set?

**Pull request merged**